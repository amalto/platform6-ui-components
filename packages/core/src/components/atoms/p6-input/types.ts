export enum P6InputType {
  email = 0,
  number = 1,
  password = 2,
  search = 3,
  tel = 4,
  text = 5,
  url = 6,
}
