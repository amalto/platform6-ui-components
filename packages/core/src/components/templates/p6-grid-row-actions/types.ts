import { IconName, IconPrefix } from '@fortawesome/fontawesome-svg-core';
import { EventEmitter } from '@stencil/core';
import { Mode } from '../../../shared/types';
import { RowId } from '../../organisms/p6-grid/core/entities';

export interface ActionsOptions {
  create: { disabled: boolean; canUse: boolean };
  view: { disabled: boolean; canUse: boolean };
  edit: { disabled: boolean; canUse: boolean };
  rename: { disabled: boolean; canUse: boolean };
  delete: { disabled: boolean; canUse: boolean };
  export: { disabled: boolean; canUse: boolean };
  [action: string]: { disabled: boolean; canUse: boolean };
}

export interface Action {
  id: keyof ActionsOptions;
  title: string;
  icon: IconName;
  iconPrefix?: IconPrefix;
  mode?: Mode;
  emitter: () => EventEmitter<{ rowId: RowId }> | undefined;
}
