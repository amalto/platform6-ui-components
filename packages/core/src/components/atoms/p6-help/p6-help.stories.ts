import { Components } from '../../../components';
import { ComponentProps, getElement, makeModeStory, makePositionStory, makeStory, Props } from '../../../shared/storybook';
import { Mode, Position } from '../../../shared/types';

const component = 'p6-help';

export default {
  title: 'Atoms/Help',
  component,
};

const componentProps: ComponentProps = ['text', 'position', 'mode'];

const getStoryField = (props?: Props<Components.P6Help>): HTMLElement => getElement(component, [], props);

export const Default = makeStory<{
  text: string;
  mode: Mode;
  position: Position;
}>({
  componentProps,
  args: {
    text: 'Tooltip',
    mode: Mode.default,
    position: Position.top,
  },
  builder: (args): HTMLElement => getStoryField(args),
});

// --- Mode
export const Modes = makeModeStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField({
      text: key,
      mode: value,
    }),
});

// --- Position
export const Positions = makePositionStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField({
      text: key,
      position: value,
    }),
});
