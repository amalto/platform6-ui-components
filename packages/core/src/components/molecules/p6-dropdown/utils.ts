export function addDropdownClass(child: Element): Element {
  if (child.nodeName === 'HR') {
    child.classList.add('dropdown-divider');
    return child;
  }

  child.classList.add('dropdown-item');
  return child;
}
