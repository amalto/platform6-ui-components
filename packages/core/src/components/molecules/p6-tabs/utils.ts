export function getTabId(id: string): string {
  return `${id}-tab`;
}

export function isTabValid(tab: Element): tab is HTMLElement {
  return 'title' in tab && 'id' in tab;
}
