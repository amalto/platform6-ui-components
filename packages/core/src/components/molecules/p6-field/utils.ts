import { isP6Control, isP6NativeControl, P6Control, P6NativeControl } from '../../../shared/form/control';
import { isInDefaultSlot } from '../../../utils/component';

// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
export function isControl(elmt: unknown): elmt is P6Control<unknown> | P6NativeControl {
  return isP6Control(elmt) || isP6NativeControl(elmt);
}

export function getControl(children: HTMLCollection): P6Control<unknown> | P6NativeControl | undefined {
  return Array.from(children).filter(isInDefaultSlot).find(isControl) as P6Control<unknown> | P6NativeControl | undefined;
}
