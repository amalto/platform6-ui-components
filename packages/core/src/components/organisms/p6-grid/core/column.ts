import { Alignment, Direction, SortOrder } from '../../../../shared/types';
import { throwError } from '../../../../utils/error';
import { Column, ColumnDefinition, ColumnId, DataItem } from './entities';

export const MIN_WIDTH = 75;
export const DEFAULT_WIDTH = MIN_WIDTH;
export const INC_WIDTH = 10;

export function defaultGetValue<T extends DataItem>(data: T, context: ColumnDefinition<T>): string {
  if (context.field === undefined || data[context.field] === undefined) {
    return throwError(`field is undefined or doesn't exist, you should provide a 'getValue' function for ${JSON.stringify(context)}`);
  }

  return `${data[context.field as keyof T]}`;
}

export function defaultSetValue<T extends DataItem>(value: string, data: T, context: ColumnDefinition<T>): T {
  if (context.field === undefined || data[context.field] === undefined) {
    return throwError(`field is undefined or doesn't exist, you should provide a 'setValue' function for ${JSON.stringify(context)}`);
  }

  return {
    ...data,
    [context.field as keyof T]: value,
  };
}

let columnIds = 0;
export function fromDefinition<T extends DataItem>(column: ColumnDefinition<T>): Column<T> {
  // eslint-disable-next-line no-plusplus
  const id = column.id || column.field || columnIds++;

  return {
    ...column,
    id: `col-${id}`,
    editable: column.editable || false,
    hidden: column.hidden || false,
    width: column.width || 100,
    align: column.align || Alignment.center,
    sortOrder: column.sortOrder || SortOrder.none,
    disableHeaderMenu: column.disableHeaderMenu || false,
    getValue: column.getValue || defaultGetValue,
    setValue: column.setValue || defaultSetValue,
  };
}

export function isHidden<T extends DataItem>(column: Column<T>): boolean {
  return !!column.hidden;
}

export function replaceColumn<T extends DataItem>(updatedColumn: Column<T>, allColumns: Column<T>[]): Column<T>[] {
  return allColumns.map(col => (col.id === updatedColumn.id ? updatedColumn : col));
}

export function move<T extends DataItem>(id: ColumnId, direction: Direction, columns: Column<T>[]): Column<T>[] {
  const from = columns.findIndex(col => col.id === id);
  const to = from + (direction === Direction.left ? -1 : 1);

  if (from < 0 || to < 0 || to >= columns.length) {
    return columns;
  }

  const result = columns.slice();
  const item = result.splice(from, 1);

  return result.slice(0, to).concat(item).concat(result.slice(to, columns.length));
}
