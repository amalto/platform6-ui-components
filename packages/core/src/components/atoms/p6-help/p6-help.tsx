import { icon } from '@fortawesome/fontawesome-svg-core';
import { Component, h, JSX, Prop } from '@stencil/core';
import { Mode, Position } from '../../../shared/types';
import { isDefaultMode, isDefaultPosition } from '../../../utils/attribute';

@Component({
  tag: 'p6-help',
  styleUrl: 'p6-help.scss',
  shadow: true,
})
export class P6Help {
  /**
   * Tooltip text
   */
  @Prop() text!: string;

  /**
   * Tooltip position (default position is top)
   */
  @Prop() position: Position = Position.top;

  /**
   * Tooltip mode
   */
  @Prop() mode: Mode = Mode.default;

  render(): JSX.Element | null {
    const img = icon({ prefix: 'fas', iconName: 'question-circle' });

    return (
      <div
        class={{
          'has-tooltip-arrow': true,
          [`has-tooltip-${Position[this.position]}`]: !isDefaultPosition(this.position, Position.top),
          [`has-tooltip-${Mode[this.mode]}`]: !isDefaultMode(this.mode),
        }}
        innerHTML={img.html[0]}
        data-tooltip={this.text}
      />
    );
  }
}
