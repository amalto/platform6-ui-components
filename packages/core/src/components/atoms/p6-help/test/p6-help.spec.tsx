import { library } from '@fortawesome/fontawesome-svg-core';
import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
import { newSpecPage } from '@stencil/core/testing';
import { enumArrayEntryToArray } from '../../../../shared/test/utils';
import { modes, positions } from '../../../../shared/types';
import { P6Help } from '../p6-help';

library.add(faQuestionCircle);

describe('p6-help', () => {
  it('default output', async () => {
    const page = await newSpecPage({
      components: [P6Help],
      html: `<p6-help text="Tooltip"></p6-help>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  describe('with position', () => {
    it.each(enumArrayEntryToArray(positions))('%s', async (key, position) => {
      const page = await newSpecPage({
        components: [P6Help],
        html: `<p6-help text="Position ${key}" position="${position}"></p6-help>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });

  describe('with mode', () => {
    it.each(enumArrayEntryToArray(modes))('%s', async (key, mode) => {
      const page = await newSpecPage({
        components: [P6Help],
        html: `<p6-help text="Mode ${key}" mode="${mode}"></p6-help>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
