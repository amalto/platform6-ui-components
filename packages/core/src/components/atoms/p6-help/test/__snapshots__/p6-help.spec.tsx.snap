// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-help default output 1`] = `
<p6-help text="Tooltip">
  <mock:shadow-root>
    <div class="has-tooltip-arrow" data-tooltip="Tooltip">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with mode danger 1`] = `
<p6-help mode="0" text="Mode danger">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-danger" data-tooltip="Mode danger">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with mode default 1`] = `
<p6-help mode="2" text="Mode default">
  <mock:shadow-root>
    <div class="has-tooltip-arrow" data-tooltip="Mode default">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with mode info 1`] = `
<p6-help mode="3" text="Mode info">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-info" data-tooltip="Mode info">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with mode primary 1`] = `
<p6-help mode="5" text="Mode primary">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-primary" data-tooltip="Mode primary">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with mode success 1`] = `
<p6-help mode="4" text="Mode success">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-success" data-tooltip="Mode success">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with mode warning 1`] = `
<p6-help mode="1" text="Mode warning">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-warning" data-tooltip="Mode warning">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with position bottom 1`] = `
<p6-help position="2" text="Position bottom">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-bottom" data-tooltip="Position bottom">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with position left 1`] = `
<p6-help position="3" text="Position left">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-left" data-tooltip="Position left">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with position right 1`] = `
<p6-help position="1" text="Position right">
  <mock:shadow-root>
    <div class="has-tooltip-arrow has-tooltip-right" data-tooltip="Position right">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;

exports[`p6-help with position top 1`] = `
<p6-help position="0" text="Position top">
  <mock:shadow-root>
    <div class="has-tooltip-arrow" data-tooltip="Position top">
      <svg aria-hidden="true" class="fa-question-circle fa-w-16 svg-inline--fa" data-icon="question-circle" data-prefix="fas" focusable="false" role="img" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
        <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z" fill="currentColor"></path>
      </svg>
    </div>
  </mock:shadow-root>
</p6-help>
`;
